lib/rules/one-var.js

```diff
@@ -273,6 +277,66 @@ module.exports = {
             return true;
         }
 
+        /**
+         * Fixer to split a VariableDeclaration into individual declarations
+         * @param   {VariableDeclaration}   declaration The `VariableDeclaration` to split
+         * @param   {?Function}             filter      Function to filter the declarations
+         * @returns {Function}                          The fixer function
+         */
+        function splitDeclarations(declaration, filter) {
+            if (!filter) {
+                filter = () => true;
+            }
+            return fixer => declaration.declarations.map(declarator => {
+                const comma = sourceCode.getTokenAfter(declarator);
+                const afterComma = sourceCode.getTokenAfter(comma, { includeComments: true });
+
+                if (filter(declarator) && comma.value === ",") {
+                    let spaceBefore = " ";
+                    let spaceAfter = "";
+
+                    if (afterComma.range[0] === comma.range[1]) {
+
+                        /*
+                         * `var x,y`
+                         * comma ^^ afterComma
+                         */
+                        spaceAfter = " ";
+                        return fixer.replaceText(comma, `;${spaceBefore}${declaration.kind}${spaceAfter}`);
+                    }
+                    if (afterComma.loc.start.line > comma.loc.end.line || afterComma.type === "Line" || afterComma.type === "Block") {
+
+                        /*
+                         * `var x,
+                         * comma ^
+                         *      y`
+                         *      ^ afterComma
+                         */
+                        spaceBefore = "\n";
+                        let lastComments = afterComma;
+                        const comments = [];
+
+                        while (lastComments.type === "Line" || lastComments.type === "Block") {
+                            comments.push(lastComments);
+                            lastComments = sourceCode.getTokenAfter(lastComments, { includeComments: true });
+                        }
+                        let replacedText = "";
+
+                        for (const comment of comments) {
+                            if (comment.type === "Line") {
+                                replacedText += `//${comment.value}\n`;
```

Comment:
we have an API: `sourceCode.getText()`, so this can be simplify to:
```js
replacedText += `${sourceCode.getText(comment)}\n`;
```

Reply:
Will fix before this gets merged